system/core/tags.php

```diff
@@ -0,0 +1,184 @@
+<?php
+	// TODO:
+	// - add styling for default theme
+	// - implement error notification if new tag exists
+	// - implement tag management page: edit, translate, remove or merge tags
+	// - implement tagcontainer pagetype
+	
+	function tagList($pageListNode, $lang) {
+		global $hyphaXml;
+		
+		// Build list of active tags for given page
+		foreach($pageListNode->getElementsByTagName('tag') as $tag) $activeTags[] = $tag->getAttribute('id');
+		
+		// Divide list of all tags stored in hypha.xml in a selected tag list and a deselected tag list
+		foreach($hyphaXml->getElementsByTagName('tag') as $tag) {
+			foreach($tag->getElementsByTagName('language') as $tagLang) {
+				if ($tagLang->getAttribute('id')==$lang) {
+					if (!empty($activeTags) && in_array($tag->getAttribute('id'), $activeTags)) $selectedTags[$tag->getAttribute('id')] = array("label" => $tagLang->getAttribute('label'), "description" => $tagLang->nodeValue);
+					else $deselectedTags[$tag->getAttribute('id')] = array("label" => $tagLang->getAttribute('label'), "description" => $tagLang->nodeValue);
+				}
+			}
+		}
+		
+		// Construct html output
+		$html = '';
+		if (!empty($activeTags) || isUser()) {
+			$html.= '<span class="prefix">' . __('tags') . ': </span>';
+			$html.= '<div id="selectedTags" style="display:inline-block;">';
+			if (!empty($selectedTags)) foreach ($selectedTags as $id => $tag) {
+				$html.= '<div id="tagSel_'.$id.'" class="tag">';
+				$html.= '<span title="'.$tag['description'].'">'.$tag['label'].'</span>';
+				if (isUser()) $html.= ' <span style="color:#8f0000; cursor:pointer;" title="'.__('remove-tag').'" onclick="deselectTag(\''.$id.'\');">⨯</span>';
```

Comment:
Please do not use inline styling, this can better be tackled in CSS. Better: 

<span class="delete-tag" />

That way the stylist can put here anything (I think)

Reply:
I would write the onclick part as:

`'onclick="deselectTag(' . htmlspecialchars(json_encode($id)).');"'`

This makes sure that any special characters will not mess up the HTML, or the javascript string literal. Note that his uses json_encode to encode a javascript string, which also generates to surrounding quotes.

Reply:
I moved all inline style to the CSS in the theme. I also changed a few "id" attributes to "class", since "id" must be unique within a page, and it does not seem unreasonable for multiple taglists to appear in one HTML page.

I also fixed the escaping.